src/pybind/mgr/cephadm/tests/test_cephadm.py

```diff
@@ -1167,6 +1170,108 @@ def test_invalid_config_option_health_warning(self, _run_cephadm, get_foreign_ce
             assert 'Ignoring invalid mgr config option test' in cephadm_module.health_checks[
                 'CEPHADM_INVALID_CONFIG_OPTION']['detail']
 
+    @mock.patch("cephadm.module.CephadmOrchestrator.get_foreign_ceph_option")
+    @mock.patch("cephadm.serve.CephadmServe._run_cephadm")
+    @mock.patch("cephadm.module.CephadmOrchestrator.set_store")
+    def test_save_devices(self, _set_store, _run_cephadm, _get_foreign_ceph_option, cephadm_module: CephadmOrchestrator):
+        _run_cephadm.side_effect = async_side_effect(('{}', '', 0))
+        entry_size = 65536  # default 64k size
+        _get_foreign_ceph_option.return_value = entry_size
+
+        class FakeDev():
+            content = {'a': 'a' * 1015}
```

Comment:
why `1015`? maybe some comment could help here :)

Reply:
just from testing, 1015 makes the resulting string when you serialize it exactly 1024 bytes, which was the goal. I can leave a comment about it.